offchain-modules/packages/x/src/handlers/ckb.ts

```
@@ -743,6 +815,75 @@ export class CkbHandler {
+      senderLockscript.hashType !== committeeMultisigLockscript.hash_type ||
+      senderLockscript.args !== committeeMultisigLockscript.args
+    ) {
+      logger.error(`sender must be committee multisig cell`);
+      return null;
+    }
+
+    for (let i = 1; i < tx.inputs.length; i++) {
+      const thisPreviousOutput = nonNullable(tx.inputs[i].previousOutput);
+      const thisPreHash = thisPreviousOutput.txHash;
+      const thisTxPrevious = await this.ckb.rpc.getTransaction(thisPreHash);
+      if (thisTxPrevious == null) {
+        return null;
+      }
+      const thisSenderLockscript = thisTxPrevious.transaction.outputs[Number(thisPreviousOutput.index)].lock;
+      if (
+        !thisSenderLockscript ||
+        !(
+          (thisSenderLockscript.codeHash !== committeeMultisigLockscript.code_hash &&
+            thisSenderLockscript.hashType !== committeeMultisigLockscript.hash_type &&
+            thisSenderLockscript.args !== committeeMultisigLockscript.args) ||
+          (thisSenderLockscript.codeHash !== collectorLockscript.code_hash &&
+            thisSenderLockscript.hashType !== collectorLockscript.hash_type &&
+            thisSenderLockscript.args !== collectorLockscript.args)
+        )
+      ) {
+        logger.error(`inputs contain must be committee multisig or collector cell`);
+        return null;
+      }
+    }
+    const unlockMemoWitness = tx.witnesses[tx.witnesses.length - 1];
+    const unlockMemo = new UnlockMemo(new Reader(unlockMemoWitness).toArrayBuffer());
+    let chain;
+    let burnIds;
+    try {
+      chain = unlockMemo.getXchain();
+      const burnIdBytes: BytesVec = unlockMemo.getBurnIds();
+      burnIds = lodash.range(burnIdBytes.length()).map((i) => new Reader(burnIdBytes.indexAt(i).raw()).serializeJson());
+    } catch (e) {
+      logger.warn(`parse recipient unlock memo data error: ${e.message} ${e.stack}`);
+      return null;
+    }
```

Don't swallow error, which brings bug here

---

fixed, try only parse unlockMemo and log error